src/components/WaveformSc/index.js

```diff
@@ -8,62 +8,52 @@ const WAVE_COLOR = '#61B25A';
 class WaveformSc extends React.Component {
 
   componentDidMount() {
-    const { activity, idx } = this.props;
+    const { activity } = this.props;
 
     if (!activity) { return; }
 
-    const { waveform_url, id } = activity;
+    const { waveform_url } = activity;
 
     if (!waveform_url) { return; }
 
     const waveformUrlJson = waveform_url.replace('.png', '.json');
 
-    const elementId = this.generateElementId(id, idx);
-
-    this.fetchJsonWaveform(elementId, waveformUrlJson);
+    this.fetchJsonWaveform(this.waveformCanvas, waveformUrlJson);
 
     // Png version will cause errors.
     // if (isPngWaveform(waveform_url)) {
     //   this.fetchPngWaveform(elementId, activity);
     // }
   }
 
-  fetchJsonWaveform(elementId, waveformUrl) {
+  fetchJsonWaveform(waveformCanvas, waveformUrl) {
     fetch(waveformUrl)
       .then(response => response.json())
       .then((data) => {
         new Waveform({
-          container: document.getElementById(elementId),
+          container: waveformCanvas,
           innerColor: WAVE_COLOR,
           data: normalizeSamples(data.samples)
         });
       });
   }
-
-  fetchPngWaveform(elementId, activity) {
-    const waveform = new Waveform({
-      container: document.getElementById(elementId),
-      innerColor: WAVE_COLOR
-    });
-    waveform.dataFromSoundCloudTrack(activity);
-  }
-
-  generateElementId(id, idx) {
-    return `waveform-${id}${idx}`;
-  }
+  // Seems like SoundCloud has switched to json instead of png.
+  // fetchPngWaveform(elementId, activity) {
+  //   const waveform = new Waveform({
+  //     container: document.getElementById(elementId),
+  //     innerColor: WAVE_COLOR
+  //   });
+  //   waveform.dataFromSoundCloudTrack(activity);
+  // }
```

Comment:
I remember that SoundCloud used json when you are not logged in, but png when you have logged in into the SoundCloud API. Or the other way around. If we can confirm that both is using json now, let's remove the Png overhead.